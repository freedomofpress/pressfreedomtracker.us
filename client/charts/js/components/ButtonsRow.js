import React from 'react'
import { Button } from './Button.js'

const labelStyle = {
  fontFamily: 'sans-serif',
  fontSize: '18px',
}

export function ButtonsRow({
  label,
  buttonLabels,
  defaultSelection = null,
  updateSelection,
  isButtonSelectable,
}) {
  const [selectedButton, setSelectedButton] = React.useState(null)

  function changeSelectedButton(buttonLabel) {
    updateSelection(selectedButton === buttonLabel ? defaultSelection : buttonLabel)
    selectedButton === buttonLabel
      ? setSelectedButton(defaultSelection)
      : setSelectedButton(buttonLabel)
  }

  return (
    <div style={{ display: 'flex', alignItems: 'center', margin: 12 }}>
      <div
        style={{
          ...labelStyle,
        }}
      >
        {label}
      </div>
      <div style={{ marginLeft: 10 }}>
        {buttonLabels.map((buttonLabel) => (
          <Button
            label={buttonLabel}
            selected={
              (defaultSelection === buttonLabel && selectedButton === null) ||
              selectedButton === buttonLabel
            }
            onClick={() => {
              changeSelectedButton(buttonLabel)
            }}
            selectable={isButtonSelectable(buttonLabel)}
            key={buttonLabel}
          />
        ))}
      </div>
    </div>
  )
}
