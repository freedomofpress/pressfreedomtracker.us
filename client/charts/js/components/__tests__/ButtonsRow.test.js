import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import ButtonsRow from '../ButtonsRow'

test('renders ButtonsRow with mocked data', () => {
	const renderer = new ShallowRenderer();
	expect(renderer.render(
		<ButtonsRow
			label="test"
			buttonLabels={['test1', 'test2']}
			updateSelection={() => {}}
			isButtonSelectable={() => true}
		/>
	)).toMatchSnapshot();
});
