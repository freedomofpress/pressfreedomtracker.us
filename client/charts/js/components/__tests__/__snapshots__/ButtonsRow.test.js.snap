// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ButtonsRow with mocked data 1`] = `
<div
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "margin": 12,
    }
  }
>
  <div
    style={
      {
        "fontFamily": "var(--font-base)",
        "fontSize": "18px",
      }
    }
  >
    test
  </div>
  <div
    style={
      {
        "marginLeft": 10,
      }
    }
  >
    <Button
      label="test1"
      onClick={[Function]}
      selectable={true}
      selected={false}
      tooltipIfUnselectable={null}
    />
    <Button
      label="test2"
      onClick={[Function]}
      selectable={true}
      selected={false}
      tooltipIfUnselectable={null}
    />
  </div>
</div>
`;
